protocol/messenger_backup_handler.go

```diff
@@ -0,0 +1,205 @@
+					errors := m.HandleBackup(messageState, msg)
+					if len(errors) > 0 {
+						for _, err := range errors {
+							m.logger.Warn("failed to handle postponed Backup messages", zap.Error(err))
+						}
+					}
+				}
+
+				m.backedupMessagesHandler.messagesToProceed = nil
+			case <-m.quit:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func (m *Messenger) HandleBackup(state *ReceivedMessageState, message protobuf.Backup) []error {
+	var errors []error
+	if m.backedupMessagesHandler.postponeHandling {
+		m.backedupMessagesHandler.addMessage(message)
+		return errors
+	}
+
+	err := m.handleBackedUpProfile(message.Profile, message.Clock)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	for _, contact := range message.Contacts {
+		err = m.HandleSyncInstallationContact(state, *contact)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	for _, community := range message.Communities {
+		err = m.handleSyncCommunity(state, *community)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+	err = m.handleBackedUpSettings(message.Setting)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if m.config.messengerSignalsHandler != nil {
+		response := wakusync.WakuBackedUpDataResponse{}
+
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyProfile, message.ProfileDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyContacts, message.ContactsDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyCommunities, message.CommunitiesDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeySettings, message.SettingsDetails)
+
+		m.config.messengerSignalsHandler.SendWakuFetchingBackupProgress(&response)
+	}
+
+	return errors
+}
+
+func (m *Messenger) handleBackedUpProfile(message *protobuf.BackedUpProfile, backupTime uint64) error {
+	if message == nil {
+		return nil
+	}
+
+	dbDisplayNameClock, err := m.settings.GetSettingLastSynced(settings.DisplayName)
+	if err != nil {
+		return err
+	}
+
+	response := wakusync.WakuBackedUpDataResponse{
+		Profile: &wakusync.BackedUpProfile{},
+	}
+
+	if dbDisplayNameClock < message.DisplayNameClock {
+		err := m.SetDisplayName(message.DisplayName, false)
+		response.AddDisplayName(message.DisplayName, err == nil)
+	}
+
+	syncWithBackedUpImages := false
+	dbImages, err := m.multiAccounts.GetIdentityImages(message.KeyUid)
+	if err != nil {
+		syncWithBackedUpImages = true
+	}
+	if len(dbImages) == 0 {
+		if len(message.Pictures) > 0 {
+			syncWithBackedUpImages = true
+		}
+	} else {
+		// since both images (large and thumbnail) are always stored in the same time, we're free to use either of those two clocks for comparison
+		lastImageStoredClock := dbImages[0].Clock
+		syncWithBackedUpImages = lastImageStoredClock < backupTime
+	}
+
+	if syncWithBackedUpImages {
+		if len(message.Pictures) == 0 {
+			err = m.multiAccounts.DeleteIdentityImage(message.KeyUid)
+			response.AddImages(nil, err == nil)
```

Comment:
Similar as above, I think we should stop executing if an unexpected error is returned